lldb/source/Target/SwiftLanguageRuntimeDynamicTypeResolution.cpp

```diff
@@ -1027,22 +1155,117 @@ SwiftLanguageRuntimeImpl::GetPromiseForTypeNameAndFrame(const char *type_name,
   lldb::addr_t metadata_location(metadata_ptr_var_sp->GetValueAsUnsigned(0));
   if (metadata_location == 0 || metadata_location == LLDB_INVALID_ADDRESS)
     return nullptr;
-
   return GetMetadataPromise(metadata_location, *metadata_ptr_var_sp);
 }
 
+static void
+ForEachGenericParameter(swift::Demangle::NodePointer node,
+                        std::function<void(unsigned, unsigned)> callback) {
+  if (!node)
+    return;
+
+  using namespace swift::Demangle;
+  switch (node->getKind()) {
+  case Node::Kind::DependentGenericParamType: {
+    if (node->getNumChildren() != 2)
+      return;
+    NodePointer depth_node = node->getChild(0);
+    NodePointer index_node = node->getChild(1);
+    if (!depth_node || !depth_node->hasIndex() || !index_node ||
+        !index_node->hasIndex())
+      return;
+    callback(depth_node->getIndex(), index_node->getIndex());
+    break;
+  }
+  default:
+    // Visit the child nodes.
+    for (unsigned i = 0; i < node->getNumChildren(); ++i)
+      ForEachGenericParameter(node->getChild(i), callback);
```

Comment:
I believe this can use range `for`

```suggestion
    for (auto *child : node)
      ForEachGenericParameter(child, callback);
```